opentelemetry-sdk/src/trace/span_processor_with_async_runtime.rs

```diff
@@ -188,13 +190,19 @@ struct BatchSpanProcessorInternal<E, R> {
     spans: Vec<SpanData>,
     export_tasks: FuturesUnordered<BoxFuture<'static, OTelSdkResult>>,
     runtime: R,
-    exporter: E,
+    exporter: Arc<RwLock<E>>,
```

Comment:
Going forward, the `SpanExporter` trait should be redesigned to use immutable references (&self) for all methods. This would allow us to remove the `RwLock` and use just `Arc<E>` for sharing the exporter across concurrent tasks - similar to how `LogExporter` is implemented.